README


This archive contains the distribution MooseX-Storage,
version 0.48:

  A serialization framework for Moose classes

This software is copyright (c) 2007 by Infinity Interactive, Inc..

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v5.020.



Changes
Revision history for MooseX-Storage

0.48      2014-10-12 02:01:31Z
  - switched the JSON serializer from JSON::Any to JSON::MaybeXS
  - interactive prompting at install time is now back!

0.47      2014-08-23 22:34:21Z
  - report the versions of all JSON and YAML backends during tests, in an
    attempt to track down an elusive bug
  - now properly serializing attributes that are set to the undefined value
    (github #6, thanks, Jason Pope!)
  - fix compatibility problem with MooseX::Role::Parameterized 1.03+ and
    MooseX::Storage::Format::JSONpm (some internal details changed that we
    were relying on)

0.46      2014-03-23 18:26:44Z
  - numerous small documentation clarifications and fixes

0.45      2013-12-22 00:48:10Z
  - new 'Value' type handler added (thanks David Golden)

0.44      2013-11-30 20:09:54Z
  - re-release to fix compile test

0.43      2013-09-11 01:47:40Z
  - removed use of deprecated Class::MOP::load_class

0.42      2013-09-08 18:00:22Z
  - all features are now configured as x_default => 1 in metadata (not yet
    supported by any cpan clients), signalling to automatically add to prereqs
    when installing non-interactively
  - re-release with properly encoded META.yml

0.41      2013-09-03 01:05:15Z
  - re-release with updated Dist::Zilla plugins to clear the dynamic_config
    flag in metadata
  - repository has moved to the GitHub Moose organization

0.40      2013-08-29 01:57:00Z
  - replaced last use of Best (held over since 2010!) to YAML::Any
  - fixed another test making use of optional features

0.39      2013-07-27 00:02:26Z
  - re-release as stable (full working Dist::Zilla conversion)

0.38      2013-07-20 16:42:03Z (TRIAL RELEASE)
  - re-release with new compile test, which properly declares all its prerequisites

0.37      2013-07-20 00:41:17Z (TRIAL RELEASE)
  - fixed compile tests so that modules using optional features are skipped
    (from 0.36-TRIAL)

0.36      2013-07-17 04:41:39Z (TRIAL RELEASE)
  - convert to Dist::Zilla, with new optional prereq handling

0.35    2013-07-15
  - properly skip optional tests when the required JSON backends aren't
    available

0.34    2013-06-20
  - JSON storage works when Cpanel::JSON::XS is the only JSON backend installed

0.33    2013-03-29
  - always JSONify data with sorted keys, to handle hash order randomization
    introduced in 5.17.* (RT#84287, Karen Etheridge)

0.32    2012-12-04
  - Change to use core Digest and Digest::SHA dists, rather than Digest::SHA1.
    RT#69811
  - Create a JSON::Any object, so setting $ENV{JSON_ANY_CONFIG} works.
    Adjust tests that were setting the wrong values and the default
    happened to work.
  - Don't rely on hash key order in tests (RT#81695)

0.31    2012-02-28
  - Add example for add_custom_type_handler to the MooseX::Storage::Engine docs. (perigrin)

  - Incorrect documentation for ->unpack method's inject parameter fixed.

  - The test suite now uses Test::Fatal instead of Test::Exception (Karen
    Etheridge).

0.30    2011-04-29
  - Make Makefile.PL features work again. RT#67170

0.29    2010-11-17
  - Remove use of Test::TempDir in favor of File::Temp (gphat)

0.28    2010-10-03
  - Make the engine_traits parameter be able to use a '+My::Trait::Name'
    prefix to specify a trait name which isn't in the
    MooseX::Storage::Engine::Trait namespace
  - Misc POD and test cleanups.

0.27    2010-04-15
  - use of parameterized roles (first added in 0.22) now works in Deferred
    Example: $object->freeze({ format => [ $role_name => \%args ] })

0.26    2010-03-25
  - Fix URI for repository in metadata.
  - Fix infinite recursion when collapsing objects which overload stringify
    to freeze themselves.

0.25    2010-03-09
  - Add support for Union types (bumps Moose dep to 0.99)

0.24    2010-01-11
  - Add additional YAML implementation tests.
  - Switch from using Best to YAML::Any.
  - Remove Test::YAML::Valid.

0.23    2009-11-10
  - Revert whitespace change in tests which breaks the tests if you
    have Test::YAML::Valid installed.

0.22    2009-11-05
  - Fix warnings when types do not have a parent type.
  - allow the use of roles outside MooseX::Storage:: (rjbs)
  - allow the use of parameterized roles (rjbs)

0.21    2009-07-14
  - Fix inconsistent dist versions with Perl::Version

0.20    2009-07-14
  - Add support for Storage( traits => [...] ) with a MooseX::Storage::Traits::XX
    namespace to alter the behaviour of the storage engine.
    (Suggestions for how to make this less fugly appreciated)
  - Add feature to disable cycle checking, either via engine trait or option
    - add docs & tests (including 1 TODO test)
  - Add feature to only serialize attributes whose predicate returns 'true'
  - Implement the unpack( $data, inject => {...} ) feature.
    - add docs & tests
  - Move Base::WithChecksum to build upon Basic.pm, avoid code duplication
  - Add extra test to ensure restoring fails if required argument isn't provided
  - Fix a bug where if a required constructor argument was not serialized, it was
    impossible to ->unpack again.

0.19    2009-07-14
  - Skip pod tests when not in author mode

0.18    2009-06-02
  - Fix test failures without JSON::Any. Patch from RT#46343 (Nicholas Clarke)
  - Added support for Maybe[...] constraints fixing RT#43165 (plu)

0.17    2009-04-06
  - Change MooseX::Storage::Engine to use get_all_attributes,
    rather than the deprecated compute_all_applicable_attributes (t0m)
  - Don't fail tests when Best is unavailable (t0m)

0.15    2008-09-29
  - MooseX::Storage
    - Remove use of deprecated alias_method routine
      which was causing issues when you used multiple
      Roles which used MooseX::Storage (t0m).

0.14    2008-06-22
  - MooseX::Storage::Engine
    - cycles are now tracked by refaddr instead
      of stringified object, which broke for any
      overloaded objects (Thanks to Jonathan Rockway)

  - Don't fail tests when YAML is unavailable


Build.PL
# This Build.PL for MooseX-Storage was generated by
# Dist::Zilla::Plugin::ModuleBuildTiny::Fallback 0.005
if (eval 'use Module::Build::Tiny 0.037; 1')
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";

    # This section for MooseX-Storage was generated by Dist::Zilla::Plugin::ModuleBuildTiny 0.006.
    use 5.008;
    # use Module::Build::Tiny 0.037;
    Build_PL();
}
else
{
    $ENV{PERL_MB_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are running this file manually, in which case you need
to learn to first fulfill all configure requires prerequisites listed in
META.yml or META.json -- or use a cpan client to install this distribution.

You can also silence this warning for future installations by setting the
PERL_MB_FALLBACK_SILENCE_WARNING environment variable, but please don't do
that until you fix your toolchain as described above.

EOW
    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));


    # This section was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.020.
    use strict;
    use warnings;

    require Module::Build; Module::Build->VERSION(0.28);


    my %module_build_args = (
      "build_requires" => {},
      "configure_requires" => {
        "ExtUtils::MakeMaker" => 0,
        "Module::Build::Tiny" => "0.037"
      },
      "dist_abstract" => "A serialization framework for Moose classes",
      "dist_author" => [
        "Chris Prather <chris.prather\@iinteractive.com>",
        "Stevan Little <stevan.little\@iinteractive.com>",
        "\x{5d9}\x{5d5}\x{5d1}\x{5dc} \x{5e7}\x{5d5}\x{5d2}'\x{5de}\x{5df} (Yuval Kogman) <nothingmuch\@woobling.org>"
      ],
      "dist_name" => "MooseX-Storage",
      "dist_version" => "0.47",
      "license" => "perl",
      "module_name" => "MooseX::Storage",
      "recommends" => {
        "IO::AtomicFile" => 0,
        "IO::File" => 0,
        "JSON::Any" => "1.15",
        "MooseX::Storage::Format::JSONpm" => 0,
        "Storable" => 0,
        "YAML" => 0,
        "YAML::Any" => 0,
        "YAML::Syck" => 0,
        "YAML::XS" => 0
      },
      "recursive_test_files" => 1,
      "requires" => {
        "Carp" => 0,
        "Data::Dumper" => 0,
        "Digest" => 0,
        "Module::Runtime" => 0,
        "Moose" => "0.99",
        "Moose::Meta::Attribute" => 0,
        "Moose::Role" => 0,
        "Scalar::Util" => 0,
        "String::RewritePrefix" => 0,
        "namespace::autoclean" => 0,
        "perl" => "5.008"
      },
      "script_files" => [],
      "test_requires" => {
        "ExtUtils::MakeMaker" => 0,
        "File::Spec" => 0,
        "File::Spec::Functions" => 0,
        "File::Temp" => 0,
        "Moose::Util::TypeConstraints" => 0,
        "Test::Deep" => 0,
        "Test::Fatal" => 0,
        "Test::More" => 0,
        "Test::Requires" => 0,
        "overload" => 0,
        "strict" => 0,
        "utf8" => 0,
        "warnings" => 0
      }
    );


    my %fallback_build_requires = (
      "ExtUtils::MakeMaker" => 0,
      "File::Spec" => 0,
      "File::Spec::Functions" => 0,
      "File::Temp" => 0,
      "Moose::Util::TypeConstraints" => 0,
      "Test::Deep" => 0,
      "Test::Fatal" => 0,
      "Test::More" => 0,
      "Test::Requires" => 0,
      "overload" => 0,
      "strict" => 0,
      "utf8" => 0,
      "warnings" => 0
    );


    unless ( eval { Module::Build->VERSION(0.4004) } ) {
      delete $module_build_args{test_requires};
      $module_build_args{build_requires} = \%fallback_build_requires;
    }

    my $build = Module::Build->new(%module_build_args);

    $build->create_build_script;
}
